hazelcast/src/main/java/com/hazelcast/map/impl/BasicMapContextQuerySupport.java

```diff
@@ -142,9 +142,10 @@ public Set queryLocalMember(String mapName, Predicate predicate, IterationType i
             if (partitionIds.isEmpty()) {
                 return result;
             }
-        } catch (QueryResultSizeExceededException e) {
-            throw ExceptionUtil.rethrow(e);
         } catch (Throwable t) {
+            if (t.getCause() instanceof QueryResultSizeExceededException) {
```

Comment:
@Donnerbart ExceptionUtil.rethrow(t); already calls getCause and throws the cause (if there is one) encapsulated in HazelastException(Runtime Exception). Can you describe more why do we need to do this change ?  Because I have a feeling you are doing the same thing with more code, which is not good.